src/python/gudhi/representations/vector_methods.py

```diff
@@ -147,7 +147,12 @@ def _grid_from_sample_range(self, X):
     if not self.keep_endpoints:
         self.new_resolution_ += self.nan_in_range_.sum()
     self.sample_range_fixed_ = _automatic_sample_range(sample_range, X)
-    self.grid_ = np.linspace(self.sample_range_fixed_[0], self.sample_range_fixed_[1], self.new_resolution_)
+    if sum([len(x) for x in X]) == 0:
```

Comment:
Hmm, actually, this test doesn't seem sufficient. If there is no NaN in `sample_range`, we don't need any points in the diagrams... I'll give it another look soon.

Reply:
Something like `if sum([len(x) for x in X]) == 0 and np.any(np.isnan(self.sample_range_fixed_))` for instance?

Reply:
I think just `if np.any(np.isnan(self.sample_range_fixed_))` is a good test that `_automatic_sample_range` failed to learn the bounds and the `linspace` branch is not going to work properly (it is too bad that `numpy.linspace(-np.inf, -np.inf, N)` doesn't return an array with N copies of `-np.inf` or we could have handled everything inside `_automatic_sample_range`). And then, if both endpoints are NaN, we can use `-np.inf`, but if one endpoint is specified, maybe we should use (N copies of) that value instead (with a warning in both cases)?

(a user explicitly asking for `sample_range=[-np.inf,-np.inf]` will still get nonsense, but I'll blame that on the user)

Reply:
> (a user explicitly asking for `sample_range=[-np.inf,-np.inf]` will still get nonsense, but I'll blame that on the user)

Actually, this suggests an alternative approach:
- in `_grid_from_sample_range`, only protect linspace with `if self.sample_range_fixed_[0] != self.sample_range_fixed_[1]`, using `np.full` in the `else` branch
- in `_automatic_sample_range`: print the warning when appropriate. Return bounds that never contain NaN (possibly (-inf,-inf), or (b,b) instead)

Reply:
I see, that makes sense. I've pushed a fix implementing this (I think).